scripts/tuple_tree_generator/tuple_tree_generator/generators/typescript.py

```diff
@@ -174,7 +174,7 @@ def get_default_value(self, field: StructField):
                 if isinstance(field.resolved_type, EnumDefinition):
                     return '"Invalid"'
                 else:
-                    return f"new {field.type}()"
+                    return f"new {field.type}({{}} as I{field.type})"
```

Comment:
@mrjackv do you approve?

Reply:
Removing it still builds fine, I'd drop it

Reply:
> Removing it still builds fine, I'd drop it

Did you try this branch? Or develop with this patch?

This matters in a case where there's a TTG-struct field inside another one.

Reply:
> Did you try this branch? Or develop with this patch?

This branch, with commit `b529e01f`, commit `76851834` gives me a C++ build error

> This matters in a case where there's a TTG-struct field inside a different struct.

Can you give me an example? Again as far as I remember, if all fields of a struct are optional then the constructor can be invoked with no arguments. Doing `{{}} as I{field.type}` would inevitably lead to bugs down the line.

Reply:
Dropped.